File: leetcode/171/Main.java
class Solution {
    public int titleToNumber(String columnTitle) {
        int sum = 0;

        for (int i = 0; i < columnTitle.length(); i++) {
            sum = sum * 26 + (columnTitle.charAt(i) - 'A' + 1);
        }
        return sum;
    }
}

class Solution1 {
    public int titleToNumber(String columnTitle) {

        int len = columnTitle.length();
        int sum = 0;
        int num = 0;
        for (int i = 0; i < len; i++) {
            // System.out.println(columnTitle.charAt(len -i -1));
            // System.out.println(columnTitle.charAt(len -i -1) - 'A' +1);
            // System.out.println(columnTitle.charAt(i));
            // System.out.println(columnTitle.charAt(i) - 'A' +1);

            num = columnTitle.charAt(i) - 'A' + 1;
            // sum += num*(26^(len-i-1));
            sum += num * (Math.pow(26, (len - i - 1)));

            // System.out.println("num = " + num);
            // System.out.println("sum = " + sum);
        }
        // return ((columnTitle.trim().length()/26)*26+columnTitle.trim().length()%26);
        // System.out.println(columnTitle.charAt(len-1));
        // System.out.println(columnTitle.charAt(len-1) - 'A' +1);

        // num = columnTitle.charAt(len-1) - 'A' +1;
        // sum += num;

        // System.out.println("num = " + num);
        // System.out.println("sum = " + sum);

        return sum;
    }
}

class Main {
    public static void main(String[] args) {
        Solution sl = new Solution();
        System.out.println(sl.titleToNumber("A"));
        System.out.println(sl.titleToNumber("AB"));
        System.out.println(sl.titleToNumber("ZY"));
    }
}

File: leetcode/171/Main.py
# class Solution(object):
#     def titleToNumber(self, columnTitle):
#         """
#         :type columnTitle: str
#         :rtype: int
#         """
#         num = 0
#         sum = 0
#         for c in columnTitle:
#             num = ord(c) - ord('A') + 1
#             sum = sum*26 + num
#         return sum

class Solution:
    def titleToNumber(self, columnTitle: str) -> int:
        num = 0
        sum = 0
        for c in columnTitle:
            num = ord(c) - ord('A') + 1
            sum = sum*26+num
        return sum


sl = Solution()
print(sl.titleToNumber("A"))
print(sl.titleToNumber("AB"))
print(sl.titleToNumber("ZY"))
